packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -90,6 +90,19 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   /// Return `null` if nothing is selected.
   SelectedContent? getSelectedContent();
 
+  /// Gets the list of selections in this object.
+  ///
+  /// The order of this list follows the same
+  /// order as the [Selectable]s contained under
+  /// this [SelectionHandler].
+  ///
+  /// Returns a list of [SelectedContentRange.empty]
+  /// when nothing is selected.
+  ///
+  /// Returns an empty list only if there is no
+  /// content under this [SelectionHandler].
+  List<SelectedContentRange> getSelections();
```

Comment:
is there a cases where the selected content range is not continuous? can we just use one range to indicate the current range?

This issue I think with this approach is that we will be tied in with how we break text.rich into smaller selectable pieces, or when we add selectioncontainer in some widgets.

Reply:
I think we can just return one range, I can't think of a case where the selection would be discontinuous. This is now changed.

Reply:
I ran into an edge case where the selection is discontinuous in the middle of collapsing the selection. This is because we send two edge update events, one to move the start and the other to move the end, so somewhere in between these two edge move operations we may have a discontinuous selection. This leaves the `SelectionDetails` in a weird state during that time, with two offsets that are not correct.

We can solve for this by creating a `CollapseSelectionEvent` which sets both edges at the same time in one event operation instead of the current two event implementation to collapse the selection. What do you think about this?

Reply:
cc @chunhtai, for any further thoughts on this thread.